src/pages/Dashboards/dashboards/Uniswap/Uniswap.js

```diff
@@ -0,0 +1,147 @@
+              </CheckProPaywall>
+            ) : (
+              <UniswapFlowBalances />
+            )}
+          </div>
+        </div>
+        <div id='uniswap_top_token_transactions'>
+          <div>
+            <UniswapTopTransactions />
+          </div>
+        </div>
+        <div id='uniswap_uni_price'>
+          <h4 className='h4 txt-b mrg-s mrg--b'>UNI Price, Age Consumed, Active Addresses (24h)</h4>
+          <p className={cx(dashboardsStyles.description, 'body-2 mrg-xxl mrg--b')}>
+            Daily active addresses signal the overall level of speculative (and utilitarian)
+            interest in a digital asset. As a result, sustained price rallies tend to necessitate a
+            strong uptick in active addresses. Spikes in Age Consumed point to a substantial amount
+            of previously idle coins moving addresses, suggesting a shift in the behavior of
+            long-term investors. These shifts are often strong indicators of upcoming price
+            volatility in either direction
+          </p>
+          <div>
+            <UniMetricsChart />
+          </div>
+        </div>
+        <div id='uniswap_token_distributor'>
+          <h4 className='h4 txt-b mrg-xxl mrg--b'>Token Distributor</h4>
+          <div>
+            {areClaimsRestricted ? (
+              <CheckProPaywall>
+                <UniswapMetrics />
+              </CheckProPaywall>
+            ) : (
+              <UniswapMetrics />
+            )}
+          </div>
+        </div>
+        <div id='uniswap_uni_token_claims'>
+          <h4 className='h4 txt-b mrg-s mrg--b'>UNI Token Claims</h4>
+          <p className={cx(dashboardsStyles.description, 'body-2 mrg-xxl mrg--b')}>
+            0x090d4613473dee047c3f2706764f49e0821d256e
+          </p>
+          <div className={styles.overviewWrapper}>
+            <UniswapHistoricalBalance
+              className={styles.balance__chart}
+              headerClassName={styles.balance__header}
+              axesTicks={BALANCE_CHART_TICKS}
+              padding={BALANCE_CHART_PADDING}
+              height={448}
+            />
+            <ChartWidget
+              height={448}
+              metrics={[Metric.uniswap_total_claims_amount, Metric.uniswap_total_claims_percent]}
+            />
+          </div>
+        </div>
+        <div id='uniswap_top_claimers'>
+          <div>
+            {areClaimsRestricted ? (
+              <CheckProPaywall>
+                <TopClaimersTable />
+              </CheckProPaywall>
+            ) : (
+              <TopClaimersTable />
+            )}
+          </div>
+        </div>
+        <div id='uniswap_uni_claims'>
+          <h4 className='h4 txt-b mrg-xxl mrg--b'>UNI Claims: Overview</h4>
+          <div>
+            {areClaimsRestricted ? (
+              <CheckProPaywall>
+                <ClaimersWidgets />
+              </CheckProPaywall>
+            ) : (
+              <ClaimersWidgets />
+            )}
```

Comment:
`CheckProPaywall` should be extended with additional property for conditional rendering, e.g. 
```
<CheckProPaywall extra={areClaimsRestricted}>
   <ClaimersWidgets />
</CheckProPaywall>
```
Every occurrence should be refactored in a similar way.